src/detect.c

```diff
@@ -805,6 +650,24 @@ static inline void DetectRulePacketRules(
 
         SCLogDebug("packet %" PRIu64 ": inspecting signature id %" PRIu32 "", p->pcap_cnt, s->id);
 
+        /* if we accept:hook'd the `packet_filter` hook, we skip the rest of the firewall rules. */
+        if (s->flags & SIG_FLAG_FIREWALL) {
+            if (skip_fw) {
+                SCLogDebug("skipping firewall rule %u", s->id);
+                goto next;
+            }
+        } else if (have_fw_rules) {
+            /* fw mode, we skip anything after the fw rules if:
+             * - flow pass is set
+             * - packet pass (e.g. exception policy) */
+            if (p->flags & PKT_NOPACKET_INSPECTION ||
+                    (pflow != NULL && pflow->flags & (FLOW_ACTION_PASS))) {
+                SCLogDebug("skipping firewall rule %u", s->id);
+                break_out_of_packet_filter = true;
```

Comment:
should not be needed to set this? Current rule is not a fw rule, so we're already beyond packet:filter